spec/support/read_only_mode_context.rb

```diff
@@ -0,0 +1,8 @@
+shared_context 'read-only mode' do
+  let(:allow_updates) { false }
+
+  it 'returns a read-only-mode-releated failure message as JSON' do
```

Comment:
I was a little confused by this. I would expect to find setup details in a shared_context, but not the test itself. Wouldn't a shared example be better for this (i.e. `it should behave like a_request_with_a_read-only_mode`)?